mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DefaultMapperContext.java

```diff
@@ -155,4 +184,12 @@ private static NameConverter buildNameConverter(Class<? extends NameConverter> c
           e);
     }
   }
+
+  private static List<MapperResultProducer> getResultProducers() {
+    ImmutableList.Builder<MapperResultProducer> result = ImmutableList.builder();
+    ServiceLoader<MapperResultProducerService> loader =
+        ServiceLoader.load(MapperResultProducerService.class);
+    loader.iterator().forEachRemaining(provider -> result.addAll(provider.getProducers()));
```

Comment:
Do we really need the `MapperResultProducerService` interface? We could load instances of `MapperResultProducer` directly using the Service Loader API.

Reply:
Indeed, I hadn't thought of that. If the order in the descriptor is deterministic (it probably is), then I'm fine loading producers directly. I'll check.

Reply:
The order is not formally defined by the `ServiceLoader` javadocs.

In practice, the Oracle JDK implementation parses the lines and puts them in a list, so they come out in the original order. But to play devil's advocate, it's not unrealistic to imagine another implementation that would use a hash set in order to eliminate duplicates (which are supposed to be skipped). 

So just to err on the safe side I think it's better to keep the top-level service interface, it doesn't add much complexity.